src/repl.ml

```diff
@@ -143,14 +158,39 @@ let get_code text_editor =
   let document = Vscode.TextEditor.document text_editor in
   if start_line = end_line && start_char = end_char then
     let line = Vscode.TextDocument.lineAt document ~line:start_line in
-    Vscode.TextLine.text line
-  else
-    Vscode.TextDocument.getText document ~range:(selection :> Vscode.Range.t) ()
+    Vscode.TextLine.range line
+  else (selection :> Vscode.Range.t)
 
 let prepare_code code =
   if String.is_suffix (String.strip code) ~suffix:";;" then code
   else code ^ ";;"
 
+(* Highlighted OCaml code *)
+let prevEvalTextDecorationType : TextEditorDecorationType.t option ref =
+  ref None
```

Comment:
to my knowledge, we don't put mutable state in a module but put it in ExtensionContext, possibly within a record that's related to repl functionality.

Reply:
Thanks for your comments. I addressed this and two following comments in the newest commit.